operators/src/source/ogr_source.rs

```diff
@@ -234,66 +197,18 @@ pub type InitializedOgrSource =
 impl VectorOperator for OgrSource {
     fn initialize(
         self: Box<Self>,
-        _context: &crate::engine::ExecutionContext,
+        context: &dyn crate::engine::ExecutionContext,
     ) -> Result<Box<crate::engine::InitializedVectorOperator>> {
-        let mut dataset_information = OgrSourceDataset::load_dataset(&self.params.layer_name)?;
-        dataset_information.project_columns(&self.params.attribute_projection);
-
-        let mut dataset = Dataset::open(&dataset_information.file_name)?;
-        let layer = dataset.layer_by_name(&dataset_information.layer_name)?;
-
-        let spatial_reference = match layer
-            .spatial_ref()
-            .and_then(|gdal_srs| gdal_srs.authority())
-        {
-            Ok(authority) => SpatialReference::from_str(&authority)?.into(),
-            Err(_) => SpatialReferenceOption::Unreferenced,
-        };
-
-        let data_type = if let Some(data_type) = dataset_information.data_type {
-            data_type
-        } else if dataset_information
-            .columns
-            .as_ref()
-            .map_or(false, |columns| columns.y.is_some())
-        {
-            // if there is a `y` column, there is no WKT in a single column
-            VectorDataType::MultiPoint
-        } else {
-            // TODO: is *data* a reasonable fallback type for an empty layer?
-            layer
-                .features()
-                .next()
-                .map_or(VectorDataType::Data, |feature| {
-                    feature
-                        .geometry_by_index(0)
-                        .map(Self::ogr_geometry_type)
-                        .unwrap_or(VectorDataType::Data)
-                })
-        };
-
-        let result_descriptor = VectorResultDescriptor {
-            spatial_reference,
-            data_type,
-        };
-
-        let default_geometry = Self::parse_default_geometry(&dataset_information, data_type)?;
-
-        if default_geometry.0.is_some() && dataset_information.on_error != OgrSourceErrorSpec::Keep
-        {
-            return Err(Error::InvalidOperatorSpec {
-                reason: "Default geometry must only be specified on error type `keep`".to_string(),
-            });
-        }
+        let info: Box<dyn LoadingInfo<OgrSourceDataset, VectorResultDescriptor>> =
+            context.loading_info(&self.params.data_set)?;
 
         Ok(InitializedOgrSource::new(
             self.params,
-            result_descriptor,
+            info.meta()?,
```

Comment:
what is a meta?

Reply:
Metadata? Yeah, I guess the naming is suboptimal.

Reply:
please find a better name. In this case its a ResultDescriptor?

Reply:
I renamed a bunch of stuff, I think it fits much better now fe0f3712025f7fc28e9c5434e3cb5dca2444a775

Reply:
ok